### ISSUE TITLE
build pdf error when rmd files contains no r code trunk

### ISSUE THREAD

**Issue by tcgriffith** [yihui/bookdown-chinese#6]
I replaced the .rmd files with my own files which contain no r trunks.While I try to build the book, gitbook(html) is ok, but pdf will report an error.

>! LaTeX Error: Environment Shaded undefined.
>Error: Failed to compile bookdown.tex. See bookdown.log for more info.
>In addition: Warning message:
>Your latexmk version seems to be too low. You may need to update the latexmk package or your LaTeX distribution. 
>Execution halted
>Exited with status 1.

After some experiments I find that if any r trunk exists in any rmd files,the  pdf will come out as expected so I don't think this is an latex problem

I'm using rstudio 1.0.13. here's my sessioninfo
__________________________
R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.1 LTS

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C               LC_TIME=en_AU.UTF-8       
 [4] LC_COLLATE=en_AU.UTF-8     LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] backports_1.0.5    bookdown_0.3.9     magrittr_1.5       rprojroot_1.2      htmltools_0.3.5   
 [6] tools_3.3.2        yaml_2.1.14        Rcpp_0.12.9        stringi_1.1.2      rmarkdown_1.3.9002
[11] knitr_1.15.1       stringr_1.2.0      digest_0.6.12      evaluate_0.10

**Comment by tcgriffith** [comment]
This [stackoverflow question](http://stackoverflow.com/questions/41052687/rstudio-pdf-knit-fails-with-environment-shaded-undefined-error) might be related but it's not answered yet.

**Comment by tcgriffith** [comment]
Solved， same problem as #1 , should we consider to comment out this line for all documents?

Solution:
> 注销掉 preamble.tex 文件 \renewenvironment{Shaded}{\begin{kframe}}{\end{kframe}}

**Comment by yihui** [comment]
I cannot reproduce the issue. Perhaps you are using an old version of LaTeX, and maybe a different version of Pandoc, too.

```r
> system('xetex --version')
XeTeX 3.14159265-2.6-0.99996 (TeX Live 2016)
kpathsea version 6.2.2
Copyright 2016 SIL International, Jonathan Kew and Khaled Hosny.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the XeTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the XeTeX source.
Primary author of XeTeX: Jonathan Kew.
Compiled with ICU version 57.1; using 57.1
Compiled with zlib version 1.2.8; using 1.2.8
Compiled with FreeType2 version 2.6.3; using 2.6.3
Compiled with Graphite2 version 1.3.8; using 1.3.8
Compiled with HarfBuzz version 1.2.6; using 1.2.6
Compiled with libpng version 1.6.21; using 1.6.21
Compiled with poppler version 0.42.0
Using Mac OS X Core Text and Cocoa frameworks

> rmarkdown::pandoc_version()
[1] ‘1.17.2’
```

**Comment by tcgriffith** [comment]
I'll make a little repo fork for demonstration

**Comment by yihui** [comment]
Oh wait, I see what you meant now. Sorry. I'll fix it in a minute.

**Comment by yihui** [comment]
Should be fixed via a3e392593b464ba31a7eceb0cd60f7e0bd112798. Thanks for the report!
